File: BackEnd/src/Model/Entities/Entity.ts
import NumberUtils from "../Utilities/NumberUtils";

abstract class Entity {
    private ID: number;

    public Entity(): void {
        this.ID = NumberUtils.UNSAVED_VALUE;
    }

    public getID(): number {
        return this.ID;
    }

    public setID(ID: number): void {
        if (this.ID != NumberUtils.UNSAVED_VALUE) {
            throw new Error("id cannot be changed");
        }

        if (ID <= NumberUtils.ZERO) {
            throw new Error("id cannot be negative or zero");
        }
        this.ID = ID;
    }
}

File: src/Model/Entities/Entity.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Interface.java to edit this template
 */
package hospiolot.model.business.entities;

import hospiolot.model.business.utilities.NumberUtils;

abstract public class Entity {
    private long ID;

    public Entity(){
        ID = NumberUtils.UNSAVED_VALUE;
    }
    
    public long getID() {
        return ID;
    }

    public void setID(long ID) {
        if(this.ID!=NumberUtils.UNSAVED_VALUE){
            throw new UnsupportedOperationException("id cannot be changed");
        }
        
        if(ID <= NumberUtils.ZERO){
            throw new IllegalArgumentException("id cannot be negative or zero");
        }
        this.ID = ID;
    }
}
